src/Pages/MembershipApplication/MembershipForm.js

```diff
@@ -38,50 +47,66 @@ export default function MembershipForm(props) {
       addon: !usernameAvailable && (
         <p className='unavailable'>User already exists!</p>
       ),
-      handleChange: e => setEmail(e.target.value)
+      ifLeftEmpty: !email && submitted && (
+        <p className='unavailable'>Email cannot be left empty</p>
+      ),
+      handleChange: e => setEmail(e.target.value),
+
     },
     {
       label: 'Password (8 or more characters)',
       type: 'password',
       addon: !passwordValid && (
         <p className='unavailable'>
-          Password requires one uppercase character and one number.
+          Password requires one uppercase character, one number
+          and at least 8 characters
+        </p>
+      ),
+      ifLeftEmpty: password.length<8 && submitted && (
+        <p className='unavailable'>
+          Password requires one uppercase character, one number
+          and at least 8 characters
         </p>
       ),
-      handleChange: e => setPassword(e.target.value)
+      handleChange: e => setPassword(e.target.value),
+
     }
   ];
 
+  function requiredFieldsEmpty() {
+    return verified && firstName && lastName &&
+    email && major && password.length >= 8;
+  }
+
   async function submitApplication() {
-    const userResponse = await checkIfUserExists(email);
-    if (userResponse.error) {
-      setUsernameAvailable(false);
-      return;
-    }
-    const registrationStatus = await registerUser({
-      firstName,
-      lastName,
-      email,
-      password,
-      major,
-      numberOfSemestersToSignUpFor: props.selectedPlan
-    });
-    if (!registrationStatus.error) {
-      sendVerificationEmail(email, firstName);
-      props.setMembershipState(memberApplicationState.CONFIRMATION);
-    } else {
-      if (registrationStatus.responseData.status === 409) {
-        window.alert('Email already exists in the system.');
-      } else if (registrationStatus.responseData.status === 400) {
-        setPasswordValid(false);
+    if(submitted===false) setSubmitted(true);
```

Comment:
could just make it`setSubmitted(true)`

Just tracing the logic of this code, shouldn't submitted only be set if `requiredFieldEmpty` is met?

Reply:
The submitted variable is meant to track if the user had clicked the submit button, then we check if the user completed all the require fields, not to track when the system submitted the application. So I changed its name to clickSubmitted instead